internal/literals/literals.go

```diff
@@ -97,74 +93,48 @@ func Obfuscate(file *ast.File, info *types.Info, fset *token.FileSet, ignoreObj
 				return true
 			}
 
-			data := make([]byte, 0, len(x.Elts))
+			data := make([]byte, 0, len(node.Elts))
 
-			for _, el := range x.Elts {
-				lit, ok := el.(*ast.BasicLit)
-				if !ok {
+			for _, el := range node.Elts {
+				elType := info.Types[el]
+
+				if elType.Value == nil || elType.Value.Kind() != constant.Int {
 					return true
 				}
-				var value byte
-				if lit.Kind == token.CHAR {
-					val, err := strconv.Unquote(lit.Value)
-					if err != nil {
-						panic(fmt.Sprintf("cannot unquote character: %v", err))
-					}
-
-					value = byte(val[0])
-				} else {
-					val, err := strconv.ParseUint(lit.Value, 0, 8)
-					if err != nil {
-						panic(fmt.Sprintf("cannot parse integer: %v", err))
-					}
-
-					value = byte(val)
+
+				value, ok := constant.Uint64Val(elType.Value)
+				if !ok {
+					return true
 				}
 
-				data = append(data, value)
+				data = append(data, byte(value))
 			}
 
 			if arrayLen > 0 {
-				cursor.Replace(withPos(obfuscateByteArray(data, arrayLen), x.Pos()))
+				cursor.Replace(withPos(obfuscateByteArray(data, arrayLen), node.Pos()))
 			} else {
-				cursor.Replace(withPos(obfuscateByteSlice(data), x.Pos()))
-			}
-
-			return true
-
-		case *ast.BasicLit:
-			switch cursor.Name() {
-			case "Values", "Rhs", "Value", "Args", "X", "Y", "Results", "Elts":
-			default:
-				return true // we don't want to obfuscate imports etc.
+				cursor.Replace(withPos(obfuscateByteSlice(data), node.Pos()))
 			}
+		}
 
-			if x.Kind != token.STRING {
-				return true
-			}
-			if len(x.Value) > maxSizeBytes {
-				return true
-			}
-			typeInfo := info.TypeOf(x)
-			if typeInfo != types.Typ[types.String] && typeInfo != types.Typ[types.UntypedString] {
-				return true
-			}
-			value, err := strconv.Unquote(x.Value)
-			if err != nil {
-				panic(fmt.Sprintf("cannot unquote string: %v", err))
-			}
+		return true
+	}
 
-			if len(value) == 0 {
-				return true
-			}
+	res := astutil.Apply(file, pre, post).(*ast.File)
 
-			cursor.Replace(withPos(obfuscateString(value), x.Pos()))
+	// some imported constants might not be needed anymore, remove unnessecary imports
+	for _, imp := range res.Imports {
+		path, _ := strconv.Unquote(imp.Path.Value)
+		if astutil.UsesImport(res, path) {
+			continue
 		}
 
-		return true
+		if !astutil.DeleteImport(fset, res, path) {
+			panic("cannot delete unused import")
```

Comment:
perhaps use fmt.Sprintf to also include context about the file and import in question